homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -133,15 +138,39 @@ class EnvoyProductionSensorEntityDescription(
         suggested_unit_of_measurement=UnitOfEnergy.MEGA_WATT_HOUR,
         suggested_display_precision=3,
         value_fn=lambda production: production.watt_hours_lifetime,
+        on_phase=None,
     ),
 )
 
 
+def phase_sensor_from_production_sensor(
+    sensors: list[EnvoyProductionSensorEntityDescription],
+) -> list[EnvoyProductionSensorEntityDescription]:
+    """Build phase sensors from production sensors."""
+    return [
+        replace(
+            sensor,
+            key=f"{sensor.key}_l{phase + 1}",
+            translation_key=f"{sensor.translation_key}_phase",
+            on_phase=PhaseNames(PHASENAMES[phase]),
+            translation_placeholders={"phase_name": f"l{phase + 1}"},
+        )
+        for sensor in sensors
+        for phase in range(0, 3)
+    ]
+
+
+PRODUCTION_PHASE_SENSORS = tuple(
```

Comment:
It would be better to store this in a `dict` by `phase` so we can simplify below

Reply:
```py
    return {
        (PhaseNames(PHASENAMES[phase])): [
            replace(
                sensor,
                key=f"{sensor.key}_l{phase + 1}",
                translation_key=f"{sensor.translation_key}_phase",
                on_phase=PhaseNames(PHASENAMES[phase]),
                translation_placeholders={"phase_name": f"l{phase + 1}"},
            )
            for sensor in sensors
        ]
        for phase in range(0, 3)
    }
    ```